src/app/components/App/Dropzone.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useCallback, useMemo} from 'react';
+import {useDropzone} from 'react-dropzone';
+import {css} from '@emotion/react';
+
+const baseStyle = css({
+  display: 'inline-block',
+  maxWidth: '400px',
+  padding: '20px',
+  marginTop: '20px',
+  borderWidth: 2,
+  borderRadius: 2,
+  borderColor: '#eeeeee',
+  borderStyle: 'dashed',
+  backgroundColor: '#fafafa',
+  color: '#bdbdbd',
+  outline: 'none',
+  transition: 'border .24s ease-in-out'
+});
+
+// const focusedStyle = css({
+//   //...baseStyle,
+//   borderColor: '#2196f3'
+// });
+
+interface DropzoneProps {
+  setFileSelected: (file: File | undefined) => void;
+  setImagePreview: (file: string) => void;
+}
+
+const Dropzone: React.FC<DropzoneProps> = ({setFileSelected, setImagePreview}) => {
+  const onDrop = useCallback((acceptedFiles) => {
+    acceptedFiles.forEach((file) => {
```

Comment:
this implementation would only work for the single file even though loops over the multiple files as would call callbacks on every file load.

Better would be either to directly use only first accepted file or change the logic to call callbacks once all of them are loaded with array of files / previews.

Reply:
as well you can set dropzone explicitly to accept only single file :)